### PR TITLE
Remove ignore-inside-blocker

### LINKED ISSUE #4918: Players can destroy grilles underneath windows
## Description

If you target it correctly, you can destroy the grille underneath the window - it's mostly focused around some edge in the inner part of the window

**Screenshots**

2021-10-16_scrot_012

### LINKED ISSUE #5089: You can select wires hidden under walls and zap yourself with them.
.

### LINKED ISSUE #5118: You can put things into containers through walls.
## Description

How to reproduce:

- Find a wall where you know there's a container on the other side. This works with most potted plants, as well as the medkits inside Chemistry and the gold toolbox in the Vault. 
- Have an object in your hand and click on the other side of the wall.
- You'll hear the object go inside. 
- To test this visibly, you can find the sweet spot and drop a lit minibomb inside.

image

Content Client_Hk6UPO1Jww

### LINKED ISSUE #5119: Items close to walls can sometimes be grabbed or deconstructed through walls.
## Description

This is an inconsistent bug that I have reproduced on local without any simulated lag. It's 100% consistent for light bulbs and light fixtures, but not for items.

How to reproduce:

- Put on a set of insulated gloves.
- Find a spot where you know there are lights on the other side of a wall. This works for both windows (such as in the below screenshots) and walls (Chemistry is a good example, I've also gotten this to work in Sec and Cargo).
- Spam clicks on the edge of the wall. You will grab the light bulbs from the sockets.
- You can also deconstruct the light fixtures, though this takes a bit of finessing.
- Occasionally, doing this with an empty hand can pick up items through walls.

I've also noticed you can attack things through walls if they're close enough to the wall. As an example, if you grab a melee weapon with decent damage it's entirely possible to destroy the Chem Dispenser/ChemMaster through the wall.

**Screenshots**

Content Client_X0qsEYmONU

Content Client_pSD93Orz8b

Content Client_BUDxqOCtko

**Additional context**

### PR DESCRIPTION
**Pull request by ElectroJr**
The main point is to fix #5118, fix #5119, fix #4918, and fix #5089 (though not the unrelated sub-floor issue) by removing the ignore-inside-blocker option from `InRangeUnobstructed`. This PR just replaces this with some component dependent predicates. In particular:
- Construction with `constructionPrototype.CanBuildInImpassable` will just ignore all anchored entities at the target location.
- If the target is an item, interactions will also ignores all anchored entities. This helps with items stuck in walls, which can happen when deconstructing wall-mounts.
- Finally, if the target has a `WallMountComponent` it will also ignore anchored entities
	- This can optionally be restricted to an arc, functionally preventing people from interacting with wall-mounts from some directions. E.g., only interact with APCs from the side of the wall that it is facing.
	- Component might need a better name. I gave windows the `WallMountComponent` as well, so that they could ignore grilles underneath them, while grilles should be properly blocked by windows. But they're not really wall-mounts.

These changes mean an entity's components determine whether some obstructions are ignored. Therefore, `InRangeUnobstructed(user, Transform(entity).MapPosition)` and `InRangeUnobstructed(user, entity)` can yield different results, because the former has no information about the target entity..

This PR will probably need some discussion as this might not be the right way to fix this. I know that during one maintainer meeting it was decided that wall mounts should not be offset from the wall, and that directional blocking would somehow be implemented. I'm not sure what the plan for that was, but I can always just remove the directional blocking from the `WallMountComponent`, at which point it might as well be turned into tag instead. Otherwise, the wall mount component is basically a temporary solution to in-wall interactions until something like anchor-layers gets added (discussed a bit in #5121).

## Static helper functions

One of the things I disliked about `InRangeUnobstructed` was that it was riddled with helper functions. For example, instead of just expecting people to take an item container and using `container.Owner` it provided 9 separate helper functions to do that for you, with even more for `Component.Owner` and `EntitCoordinate` -> `MapCoordinate`. Many of these variants went unused, and even when the static functions were used they were often just obscuring the use of `IoCManager.Resolve()` and `EntitySystem.Get()` calls. Really, the calling systems should just have been directly using the a `[Dependency] SharedInteractionSystem`. So instead of bothering to update all of these helper functions I've just removed the static helpers altogether. If people really value them I can add them back in.

## `WallMountComponent` uses.

- I added `WallMountComponent` to a bunch of prototypes, but probably missed some.
- Some of these I gave no angle restrictions (e.g. fire extinguisher cabinet or glass windows).
	- Without directional occlusion this would probably just be confusing, seeing as the entities appear to be accessible from any direction
- Others entities, like APCs and fire-alarms, were given angle-restrictions, requiring the user to be "south" of the entity.
	- Should be fine, as having these entities visually rotated / upside down doesn't look out of place.
	- This might require some maps to be updated if they weren't rotated.

:cl:
- fix: Fixed being able to interact with entities through walls when the station has a ~45 degree rotation.
- tweak: Some wall-mounted entities now require the user to to be on a specific side of the wall in order to interact with them.

### CODE REVIEW

**Comment by mirrorcult** [comment]
worlds most based PR

File: Content.Shared/Interaction/SharedInteractionSystem.cs [hunk 1]
```diff
@@ -433,55 +366,92 @@ protected bool ValidateInteractAndFace(EntityUid user, EntityCoordinates coordin
...
+                    || (predicate?.Invoke(e) ?? false)
+                    || (wallPredicate?.Invoke(e) ?? false);
+            };
+
+            var inRange = InRangeUnobstructed(origin, mapPos, range, collisionMask, combinedPredicate, popup);
+
+            if (!inRange && popup)
+            {
+                var message = Loc.GetString("interaction-system-user-interaction-cannot-reach");
+                origin.PopupMessage(message);
```
Comment (metalgearsloth) [inline Content.Shared/Interaction/SharedInteractionSystem.cs:390]: Does popupsystem have a method for this?

**Review by metalgearsloth — APPROVED** [review]
Wallmount's probably the best strategy in terms of having interactions able to tell the difference between an embedded item and something like a cabinet that's directional.

File: Content.Shared/Interaction/SharedInteractionSystem.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Shared/Interaction/SharedInteractionSystem.cs:390]: yeah it does. This is the problem with proxy methods. So convenient yet they hide all the resolves & gets.

**Pull request merged**